logpolicy/logpolicy.go

```diff
@@ -101,9 +102,9 @@ func (l logWriter) Write(buf []byte) (int, error) {
 // logsDir returns the directory to use for log configuration and
 // buffer storage.
 func logsDir() string {
-	systemdCacheDir := os.Getenv("CACHE_DIRECTORY")
-	if systemdCacheDir != "" {
-		return systemdCacheDir
+	systemdStateDir := os.Getenv("STATE_DIRECTORY")
```

Comment:
is it necessarily systemd? or is this defined by XDG or something?

Reply:
Currently, it's systemd only. Systemd doesn't follow XDG on this because XDG is for locations that apply to users on a desktop, instead it defines its own FOO_DIRECTORY for a half dozen or so foo's that are relevant to the service lifecycle (e.g. state, cache, runtime, temp, ...). Basically this fires for all our linux packages, and nowhere else - hence why it's optional.